src/control/lib/control/pool.go

```diff
@@ -51,21 +51,26 @@ func checkUUID(uuidStr string) error {
 	return errors.Wrapf(err, "invalid UUID %q", uuidStr)
 }
 
-// formatNameGroup converts system names to principal and if both user and group
-// are unspecified, takes effective user name and that user's primary group.
+// formatNameGroup converts system names to principal and ensures both user and
+// group are, takes effective user name and that user's primary group.
```

Comment:
The second part of the comment got a bit mixed up. Maybe something like "formatNameGroup converts system names to principals. If user or group isn't provided, the effective user and/or effective group will be used."